### PR TITLE
GUI: Adds delay to blur to reduce render cycles

### LINKED ISSUE #292: GUI: Blur feature too CPU intensive
When we adjust blur via slider, the number of render cycles is too much.
Change the slider to call `blur` method only `onmouseup`

### PR DESCRIPTION
**Pull request by nkprince007**
Fixes #292

### CODE REVIEW

**Comment by abhsag24** [comment]
Hi thank you for your Patch make sure you have the right commit structure that is 
```
tag: Shortlog

body
Fixes <issue number/url>
```

File: uiplib/gui/gallery.py [hunk 1]
```diff
@@ -35,10 +36,11 @@ def set_image(self, imagePath):
             self.cv = Canvas(self, width=self.width, height=self.height)
             self.cv.pack(fill=BOTH, expand=YES)
         if not self.slider:
+            self._job = None
```
Comment (abhsag24) [inline uiplib/gui/gallery.py]: when is this not `None` ?

File: uiplib/gui/gallery.py [hunk 2]
```diff
@@ -50,7 +52,12 @@ def show_image(self, image):
...
         self.cv.create_image(0, 0, anchor="nw", image=self.tk_image)
 
-    def blur_image(self, val):
+    def _blur_helper(self, event):
+        if self._job:
+            self.appObj.after_cancel(self._job)
+        self._job = self.appObj.after(250, self.blur_image)
+
+    def blur_image(self):
         """Blur chosen image by val amount."""
```
Comment (abhsag24) [inline uiplib/gui/gallery.py]: change docstring

File: uiplib/gui/gallery.py (hunk 1 above)
Comment (nkprince007) [inline uiplib/gui/gallery.py]: In line 58

File: uiplib/gui/gallery.py [hunk 3]
```diff
@@ -50,7 +52,13 @@ def show_image(self, image):
...
         self.cv.create_image(0, 0, anchor="nw", image=self.tk_image)
 
-    def blur_image(self, val):
-        """Blur chosen image by val amount."""
-        show = self.image.blur(val)
+    def _blur_helper(self, event):
+        """Helper to schedule the blur_image task."""
+        if self._job:
+            self.appObj.after_cancel(self._job)
+        self._job = self.appObj.after(250, self.blur_image)
```
Comment (abhsag24) [inline uiplib/gui/gallery.py]: if there is no `job` why wait 250 ms?

Comment (Aniq55) [inline uiplib/gui/gallery.py]: I think time.sleep(<time>) is better

Comment (abhsag24) [inline uiplib/gui/gallery.py]: @Aniq55 no after is a tkinter's inhouse function i think letting them handle is better than self imposed `time.sleep()`

Comment (abhsag24) [inline uiplib/gui/gallery.py]: plus after returns a `job` which we can check later `time.sleep()` is very basic and a very hacky way to apply here

Comment (nkprince007) [inline uiplib/gui/gallery.py]: Yeah, that's a point @abhsag24, will fix that.

Comment (nkprince007) [inline uiplib/gui/gallery.py]: But, that's again making it too CPU intense @abhsag24 We need it to wait for some time before it does anything. It seems to be ideal at 200-250.

Comment (nkprince007) [inline uiplib/gui/gallery.py]: Reducing that hogs the CPU time again.

Comment (abhsag24) [inline uiplib/gui/gallery.py]: yes :P i think this is fine :D

**Pull request merged**